src/dashboard/apps/consent/helpers.py

```diff
@@ -76,3 +83,65 @@ def send_mail(recipients: list, template_id: int, email_data: dict) -> int | Non
         # fail silently and send a sentry log
         sentry_sdk.capture_exception(e)
         return 0
+
+
+def duplicate_expiring_consents():
+    """Duplicate expiring consents to ensure continuity.
+
+    Perform the duplication of expiring consents to ensure continuity for consents that
+    are nearing their expiration date and have not been duplicated yet.
+    The function identifies consents that are set to expire shortly and generates
+    new consents starting immediately after the expiry date of the current consent,
+    with an extended validity period.
+
+    - Consents should be duplicated if their end date is earlier than or equal to
+    `now + CONSENT_NUMBER_DAYS_END_DATE`.
+    - duplicated consent date:
+        - start date of new consent is end date of current consent
+        - end date of new consent is calculate from CONSENT_NUMBER_DAYS_END_DATE
+
+    Returns:
+        List[Consent]: A list of newly created Consent objects that have been duplicated
+         from the expiring consents.
+    """
+    # calculate the date from which to check
+    checking_date = consent_checking_date()
+
+    # calculate the end date of new duplicate consents
+    duplicate_end_date = duplicate_consent_end_date()
```

Comment:
Maybe it would be more explicit to have the dates logic in here instead of an helper?

Reply:
The advantage of keeping it in an external function is the ability to perform unit tests on it. What do you think about moving this function directly into this module?

Reply:
Good point. Yes!